internal/releasesources/release_sources.go

```diff
@@ -0,0 +1,468 @@
+			continue
+		}
+
+		if apiRelease.Default && configurationHasDefaultRelease {
+			apiRelease.Default = false
+		}
+
+		mergedReleaseImages = append(mergedReleaseImages, apiRelease)
+	}
+
+	return mergedReleaseImages
+}
+
+func (h *ReleaseSourcesHandler) syncReleaseImages(tx *gorm.DB) error {
+	apiReleaseImages, err := h.getReleaseImagesFromAPI()
+	if err != nil {
+		return err
+	}
+	err = h.setDefaultRelease(apiReleaseImages)
+	if err != nil {
+		return err
+	}
+	supportLevels, err := h.syncOpenshiftVersionsSupportLevels(tx)
+	if err != nil {
+		return err
+	}
+	err = h.setSupportLevels(apiReleaseImages, supportLevels)
+	if err != nil {
+		return err
+	}
+	configurationReleaseImages := h.getConfigurationReleaseImages()
+	err = h.setSupportLevels(configurationReleaseImages, supportLevels)
+	if err != nil {
+		return err
+	}
+	apiReleaseImages = h.removeReleaseImageDuplicates(apiReleaseImages)
+	releaseImages := h.mergeReleaseImages(configurationReleaseImages, apiReleaseImages)
+
+	// Deleting all releases before adding again in order to
+	// store only the releases according to RELEASE_SOURCES
+	err = h.deleteAllReleases(tx)
+	if err != nil {
+		return err
+	}
+
+	err = h.insertToDB(tx, releaseImages)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *ReleaseSourcesHandler) createSupportLevels(supportLevelGraph *SupportLevelGraph) []*common.OpenshiftVersionSupportLevel {
+	mapAPISupportLevelToOurSupportLevel := map[string]string{
+		"End of life":         models.OpenshiftVersionSupportLevelEndOfLife,
+		"Maintenance Support": models.OpenshiftVersionSupportLevelMaintenance,
+		"Full Support":        models.OpenshiftVersionSupportLevelProduction,
+	}
+
+	supportLevels := []*common.OpenshiftVersionSupportLevel{}
+
+	for _, Data := range supportLevelGraph.Data {
+		for _, version := range Data.Versions {
+			if convertion, ok := mapAPISupportLevelToOurSupportLevel[version.Type]; ok {
+				version.Type = convertion
+			}
+			supportLevels = append(
+				supportLevels,
+				&common.OpenshiftVersionSupportLevel{OpenshiftVersion: version.Name, SupportLevel: version.Type},
+			)
+		}
+	}
+
+	return supportLevels
+}
+
+func (h *ReleaseSourcesHandler) syncOpenshiftVersionsSupportLevels(tx *gorm.DB) ([]*common.OpenshiftVersionSupportLevel, error) {
+	supportLevelGraph, err := h.supportLevelClient.GetSupportLevels(h.config.OpenshiftMajorVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	supportLevels := h.createSupportLevels(supportLevelGraph)
+	if err = tx.Where("TRUE").Delete(&common.OpenshiftVersionSupportLevel{}).Error; err != nil {
+		h.log.WithError(err).Debug("error occurred while trying to delete suppurt levels table")
+		return nil, err
+	}
+
+	if err = tx.Create(&supportLevels).Error; err != nil {
+		h.log.WithError(err).Debug("error occurred while trying to create suppurt levels table")
+		return nil, err
+	}
+
+	return supportLevels, nil
+}
+
+// All candidate/pre-release releases are considered support level 'beta'.
```

Comment:
function doc comments should follow standard go function doc format (e.g. start with the name of the function)